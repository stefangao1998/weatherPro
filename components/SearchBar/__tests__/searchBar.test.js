import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {create} from 'react-test-renderer';

import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

configure({adapter: new Adapter()});

import {SearchBar} from '../index';

describe('SearchBar component', () => {
  it('should render without issues', () => {
    const component = shallow(<SearchBar />, {context: ''});

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with tabbar', () => {
    const component = shallow(
      <SearchBar 
        withTabBar={false}
      />,
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without cancel Button', () => {
    const component = shallow(
      <SearchBar 
        withTabBar={false}
        onCancel={false}
      />,
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });


  it('should have onPress event', () => {
    const onPress = jest.fn();
    const component = shallow(<SearchBar onPress={()=>onPress}/>);
    expect(toJson(component)).toMatchSnapshot();
  });

});
