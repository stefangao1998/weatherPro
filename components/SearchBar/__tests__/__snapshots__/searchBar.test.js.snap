// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should have onPress event 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4DDBC8",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 8,
          "marginHorizontal": 16,
          "marginTop": 33,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 5,
            "flex": 7,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "space-between",
            "shadowColor": "#E1E5E4",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
            "width": 307,
          }
        }
      >
        <Icon
          name="search"
          size={18}
          style={
            Object {
              "color": "#636F8B",
              "flex": 1,
              "marginLeft": 10,
            }
          }
        />
        <Component
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="搜索趣团码，立即加入好友活动"
          rejectResponderTermination={true}
          returnKeyType="search"
          style={
            Object {
              "flex": 10,
              "textAlign": "left",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Icon
          name="close-circle"
          onPress={[Function]}
          size={18}
          style={
            Object {
              "color": "#E2E5EB",
              "flex": 1,
              "marginRight": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ForwardRef>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
              }
            }
          >
            取消
          </Text>
        </ForwardRef>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "elevation": 4,
        "flexDirection": "row",
        "height": 48,
        "marginBottom": 8,
        "shadowColor": "#9EABC0",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#1C2739",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        趣团码
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#4DDBC8",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#8E9DB6",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        达人
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#8E9DB6",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        活动
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`SearchBar component should render with tabbar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4DDBC8",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 8,
          "marginHorizontal": 16,
          "marginTop": 33,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 5,
            "flex": 7,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "space-between",
            "shadowColor": "#E1E5E4",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
            "width": 307,
          }
        }
      >
        <Icon
          name="search"
          size={18}
          style={
            Object {
              "color": "#636F8B",
              "flex": 1,
              "marginLeft": 10,
            }
          }
        />
        <Component
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="搜索趣团码，立即加入好友活动"
          rejectResponderTermination={true}
          returnKeyType="search"
          style={
            Object {
              "flex": 10,
              "textAlign": "left",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Icon
          name="close-circle"
          onPress={[Function]}
          size={18}
          style={
            Object {
              "color": "#E2E5EB",
              "flex": 1,
              "marginRight": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ForwardRef>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
              }
            }
          >
            取消
          </Text>
        </ForwardRef>
      </View>
    </View>
  </View>
</View>
`;

exports[`SearchBar component should render without cancel Button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4DDBC8",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 8,
          "marginHorizontal": 16,
          "marginTop": 33,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 5,
            "flex": 7,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "space-between",
            "shadowColor": "#E1E5E4",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
            "width": 307,
          }
        }
      >
        <Icon
          name="search"
          size={18}
          style={
            Object {
              "color": "#636F8B",
              "flex": 1,
              "marginLeft": 10,
            }
          }
        />
        <Component
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="搜索趣团码，立即加入好友活动"
          rejectResponderTermination={true}
          returnKeyType="search"
          style={
            Object {
              "flex": 10,
              "textAlign": "left",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Icon
          name="close-circle"
          onPress={[Function]}
          size={18}
          style={
            Object {
              "color": "#E2E5EB",
              "flex": 1,
              "marginRight": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ForwardRef
          onPress={false}
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
              }
            }
          >
            取消
          </Text>
        </ForwardRef>
      </View>
    </View>
  </View>
</View>
`;

exports[`SearchBar component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4DDBC8",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 8,
          "marginHorizontal": 16,
          "marginTop": 33,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 5,
            "flex": 7,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "space-between",
            "shadowColor": "#E1E5E4",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
            "width": 307,
          }
        }
      >
        <Icon
          name="search"
          size={18}
          style={
            Object {
              "color": "#636F8B",
              "flex": 1,
              "marginLeft": 10,
            }
          }
        />
        <Component
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="搜索趣团码，立即加入好友活动"
          rejectResponderTermination={true}
          returnKeyType="search"
          style={
            Object {
              "flex": 10,
              "textAlign": "left",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Icon
          name="close-circle"
          onPress={[Function]}
          size={18}
          style={
            Object {
              "color": "#E2E5EB",
              "flex": 1,
              "marginRight": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ForwardRef>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
              }
            }
          >
            取消
          </Text>
        </ForwardRef>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "elevation": 4,
        "flexDirection": "row",
        "height": 48,
        "marginBottom": 8,
        "shadowColor": "#9EABC0",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#1C2739",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        趣团码
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#4DDBC8",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#8E9DB6",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        达人
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#8E9DB6",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 22,
            "marginBottom": 7,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        活动
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": -10,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
